lib/rules/no-unused-vars.js

```diff
@@ -178,6 +191,47 @@ module.exports = {
             };
         }
 
+        /**
+         * Generate the warning message about a variable being used even though
+         * it is marked as being ignored.
+         * @param {Variable} usedIgnoredVar eslint-scope variable object
+         * @returns {UsedIgnoredVarMessageData} The message data to be used with
+         * this used ignored variable.
+         */
+        function getUsedIgnoredMessageData(usedIgnoredVar) {
+            const def = usedIgnoredVar.defs && usedIgnoredVar.defs[0];
+            let additional = "";
+
+            if (def) {
+                const defType = def.type;
+                let type;
+                let pattern;
+
+                if (defType === "CatchClause" && config.caughtErrorsIgnorePattern) {
+                    type = "args";
+                    pattern = config.caughtErrorsIgnorePattern.toString();
+                } else if (defType === "Parameter" && config.argsIgnorePattern) {
+                    type = "args";
+                    pattern = config.argsIgnorePattern.toString();
+                } else if (defType !== "Parameter" && config.varsIgnorePattern) {
+                    type = "vars";
+                    pattern = config.varsIgnorePattern.toString();
+                } else if (def.name.parent.type === "ArrayPattern" && config.destructuredArrayIgnorePattern) {
+                    type = "elements of array destructuring";
+                    pattern = config.destructuredArrayIgnorePattern.toString();
+                }
```

Comment:
```suggestion
                switch (true) {
                    case defType === "CatchClause" && config.caughtErrorsIgnorePattern:
                        type = "args";
                        pattern = config.caughtErrorsIgnorePattern.toString();
                        break;
                    case defType === "Parameter" && config.argsIgnorePattern:
                        type = "args";
                        pattern = config.argsIgnorePattern.toString();
                        break;
                    case defType !== "Parameter" && config.varsIgnorePattern:
                        type = "vars";
                        pattern = config.varsIgnorePattern.toString();
                    case def.name.parent.type === "ArrayPattern" && config.destructuredArrayIgnorePattern:
                        type = "elements of array destructuring";
                        pattern = config.destructuredArrayIgnorePattern.toString();
                        break;
                }
```

Reply:
This seems like an anti pattern.

Also, there are 2 other functions that are very similar to this one in this rule already so making this change would be inconsistent with how the rest of the rule is formatted.

Reply:
This is clearly not an anti pattern, as TypeScript 5.3 [just added](https://devblogs.microsoft.com/typescript/announcing-typescript-5-3-beta/#switch-true-narrowing#switch-true-narrowing) type narrowing for this pattern. I feel it resembles the powerful `match` present usually in functional languages.

This being said, I agree with you that we should keep the same code style and not mix them. It was just a suggestion.